src/VisualStudio/IntegrationTest/TestUtilities/VisualStudioInstance.cs

```diff
@@ -144,8 +141,25 @@ public T ExecuteInHostProcess<T>(Type type, string methodName)
         public void ActivateMainWindow(bool skipAttachingThreads = false)
             => _inProc.ActivateMainWindow(skipAttachingThreads);
 
-        public void WaitForApplicationIdle()
-            => _inProc.WaitForApplicationIdle();
+        public void WaitForApplicationIdle(CancellationToken cancellationToken)
+        {
+            using (var cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken))
+            {
+                try
+                {
+                    Task.WhenAll(
```

Comment:
Seems like .Wait(cancellationToken) is exactly what you want here with many fewer lines of code.

Reply:
➡️ Fixed